[Bug] Telegram always enters recovery loop and causes getUpdates 409 Conflict (v1.2.2-beta.5)

## Description

After upgrading to v1.2.2-beta.5, the Telegram platform **always** enters `platform recovery loop` on startup instead of starting normally, causing continuous `409 Conflict: terminated by other getUpdates request` errors. Only a single cc-connect instance is running.

This is **different from #212** — there is no `setMyCommands` failure or `BOT_COMMANDS_TOO_MUCH` error. Command registration succeeds (41 commands), yet the conflict persists.

## Environment

- cc-connect: v1.2.2-beta.5 (commit 28513f3, built 2026-03-31)
- OS: macOS arm64 (Darwin 25.3.0)
- Platforms configured: weixin + telegram
- Running via launchd (KeepAlive)
- Single instance confirmed via `pgrep`

## Steps to Reproduce

1. Configure `config.toml` with weixin and telegram platforms
2. Start cc-connect (via launchd or foreground)
3. Observe logs — Telegram enters recovery loop every time

## Logs (sensitive info redacted)

```
level=INFO  msg="config loaded" path=config.toml
level=INFO  msg="session: loaded from disk" sessions=3
level=INFO  msg="speech: enabled" provider=qwen
level=INFO  msg="platform ready" project=myproject platform=weixin
level=INFO  msg="platform recovery loop started" project=myproject platform=telegram
level=WARN  msg="engine started with partial readiness" project=myproject agent=claudecode ready=1 pending=1 failed=0
level=INFO  msg="cron: scheduler started" jobs=0
level=INFO  msg="api server started" socket=...
level=INFO  msg="cc-connect is running" projects=1
level=INFO  msg="telegram: connected" bot=my_bot
level=INFO  msg="platform ready" project=myproject platform=telegram
level=INFO  msg="telegram: registered bot commands" count=41
level=INFO  msg="[TGBOT] [ERROR] error get updates, conflict, Conflict: terminated by other getUpdates request; make sure that only one bot instance is running"
level=INFO  msg="[TGBOT] [ERROR] error get updates, conflict, ..."
(repeats every ~5-10 seconds indefinitely)
```

## What I've verified

1. **Single instance**: `pgrep` confirms only one cc-connect process on the machine. No other device or server is running cc-connect with this bot token.
2. **No webhook**: `getWebhookInfo` returns `url: ""`, no webhook is set.
3. **deleteWebhook + drop_pending_updates**: Tried calling the Telegram API to clear state before starting — conflict persists.
4. **Full stop + wait 60s + restart**: Stopped the service, waited over 60 seconds for any lingering long-poll to expire, then restarted — conflict persists immediately.
5. **Disabled weixin platform**: Commented out weixin platform config so only telegram remains. Telegram **still** enters recovery loop (`ready=0 pending=1`) and still conflicts. So this is not caused by multi-platform interaction.
6. **setMyCommands succeeds**: Unlike #212, there is no command registration failure. All 41 commands register successfully.

## Analysis

The key observation is that Telegram **always** enters `platform recovery loop` instead of starting normally. The recovery loop appears to create a second `getUpdates` goroutine that conflicts with the primary one — the bot is fighting itself.

This may be related to the `go-telegram/bot` library migration mentioned in the v1.2.2-beta.5 release notes. I have not tested whether v1.2.2-beta.4 (which used the old Telegram library) has the same issue.

## Expected

Telegram platform should start normally (not via recovery loop) and process messages without self-conflicting `getUpdates` calls.

--- a/platform/telegram/telegram.go
+++ b/platform/telegram/telegram.go
@@ -223,7 +223,43 @@ func defaultNewBot(token string, onUpdate func(context.Context, *models.Update),
 	if err != nil {
 		return nil, nil, nil, fmt.Errorf("getMe: %w", err)
 	}
+
+	// Drain pending updates before starting polling to avoid 409 Conflict.
+	// This clears any outstanding long-poll request from a previous instance.
+	drainPendingUpdates(token, httpClient)
+
 	return b, me, b.Start, nil
+}
+
+// drainPendingUpdates clears any pending updates on Telegram's side by calling
+// getUpdates with offset=-1. This terminates any outstanding long-poll request
+// from a previous bot instance, preventing 409 Conflict errors on restart.
+func drainPendingUpdates(token string, httpClient *http.Client) {
+	apiURL := "https://api.telegram.org/bot" + token + "/getUpdates?offset=-1&timeout=0"
+	client := httpClient
+	if client == nil {
+		client = http.DefaultClient
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, apiURL, nil)
+	if err != nil {
+		slog.Debug("telegram: drain updates request creation failed", "error", err)
+		return
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		slog.Debug("telegram: drain updates request failed", "error", err)
+		return
+	}
+	defer resp.Body.Close()
+
+	// Read and discard response body to ensure connection is properly closed
+	_, _ = io.Copy(io.Discard, resp.Body)
+
+	slog.Debug("telegram: drained pending updates", "status", resp.StatusCode)
 }
 
 func (p *Platform) connectLoop(ctx context.Context) {

